populate.go

```diff
@@ -0,0 +1,69 @@
+// Copyright (c) 2017 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package fx
+
+import (
+	"fmt"
+	"reflect"
+)
+
+// Populate sets values with values from the dependency injection container
+// before application start. All arguments must be pointers to the values
+// that must be populated. Pointers to structs that embed In are supported,
+// which can be used to populate multiple values in a struct.
+func Populate(targets ...interface{}) Option {
```

Comment:
Pleas add usage samples or example tests before merging. We need to
demonstrate basic usage plus usage with `fx.In`s.

Reply:
worth mentioning that it mirrors `go.uber.org/config.Provider.Populate`? or nah?

Reply:
Good idea, adding a simple example test, let me know if you want to make it more complex or if it's fine.

The intention was to keep this similar to `Populate`, but not sure it's worth calling that out (wanted a consistent experience, but they don't really need to be cross-referenced).